### PR TITLE
added a refresh game directory option to the file menu

### PR DESCRIPTION
**Pull request by jfmherokiller**

### CODE REVIEW

File: src/citra_qt/main.ui [hunk 1]
```diff
@@ -45,7 +45,7 @@
      <x>0</x>
      <y>0</y>
      <width>1081</width>
-     <height>19</height>
+     <height>22</height>
```
Comment (jfmherokiller) [inline src/citra_qt/main.ui]: I dont know why this changed

**Comment by yuriks** [comment]
Is there any particular demand for this feature? In all cases you can simply re-open Citra. I don't think it's worth the UI clutter.

A better way of doing something like this would be to use [QFileSystemWatcher](http://doc.qt.io/qt-5/qfilesystemwatcher.html) to automatically detect changes to game directories and update the list.

**Comment by jfmherokiller** [comment]
@yuriks well i thought it was just a good idea. ill see if i can implement it using what you described instead of a ui option and add it to this pull request

**Comment by jfmherokiller** [comment]
@yuriks i was successfully able to implement it im just not sure if my organization is very good

**Comment by jroweboy** [comment]
Maybe I'm misreading the code, but it doesn't look like it handles the following case

Open citra -> set game folder directory -> close citra
Open citra -> add a new game to the folder -> no refresh?

I don't see where the watcher adds the games list directory onload. It looks like it only adds it when the Select Game Directory is clicked.

**Comment by yuriks** [comment]
Also, AFAIK watching directories isn't recursive, so you need to actually watch all it's sub-diretories and keep track of them as they're added and removed. (It's probably easier to just clear the watch list and do a full re-scan instead of attempting to scan only changed files and dirs.)

**Comment by mailwl** [comment]
or, as a variant - just leave a refresh button

**Comment by jroweboy** [comment]
@mailwl it seems kinda silly to have a menu item to refresh the game list
when QT has as built in file watcher to handle the hard part of
automatically watching the folder in a cross platform way.

Is there any compelling reason *to* have it as a menu option besides being
marginally easier to code it that way? Having a menu option means its more
clutter in the UI and its more surprising to users ("wait i have to click
refresh?"). Adding a file watcher is much more user friendly without being
much more complicated to code.

**Comment by mailwl** [comment]
@jroweboy as you wish :)

**Comment by jfmherokiller** [comment]
jesus this thing is more in depth then i thought lol

**Comment by jroweboy** [comment]
@jfmherokiller any plans on finishing this? if you need some help you can hop on IRC or discord and I wouldn't mind offering some advice. If you don't want to let me know because i think its a great idea and would love to see it in citra.

**Comment by jfmherokiller** [comment]
@jroweboy the only reason i stopped is because i dont really know enough about qt and c++ in general i kinda make up stuff on the fly and its not very good

**Pull request closed without merging**